test/e2e/atlas/kubernetes_config_generate_test.go

```diff
@@ -1052,52 +1061,84 @@ func TestProjectWithStreamsProcessing(t *testing.T) {
 			}
 
 			if connection, ok := object.(*akov2.AtlasStreamConnection); ok {
-				assert.Equal(
-					t,
-					&akov2.AtlasStreamConnection{
-						TypeMeta: metav1.TypeMeta{
-							Kind:       "AtlasStreamConnection",
-							APIVersion: "atlas.mongodb.com/v1",
-						},
-						ObjectMeta: metav1.ObjectMeta{
-							Name: resources.NormalizeAtlasName(
-								fmt.Sprintf("%s-%s-%s", generator.projectName, generator.streamInstanceName, generator.streamConnectionName),
-								resources.AtlasNameToKubernetesName(),
-							),
-							Namespace: targetNamespace,
-							Labels: map[string]string{
-								"mongodb.com/atlas-resource-version": "2.4.0",
+				if connection.Spec.Name == sampleStreamName {
```

Comment:
Why do you need this conditional check? Don't they always come in the same order?
also, I think `assert` has a `MatchElements()` that check slices independently of order

Reply:
We're receiving more objects from `getK8SEntities` than we're checking for (like the Project), which means using assert's ElementsMatch doesn't work unless we add everything else to this list.

We need to do this check because we're just looping over this list and checking for a stream connection, which previously we assumed would be the only one. Now it could be any one of the two (one which we created, one which is always there).